### PR TITLE
add new provider for Algolia Places

### PR DESCRIPTION
**Pull request by Fabex**
New Provider to use Algolia Places

## TODO
 - [x] Make Tests

### CODE REVIEW

File: src/Provider/AlgoliaPlaces/AlgoliaPlaces.php [hunk 1]
```diff
@@ -0,0 +1,223 @@
...
+    const TYPE_BUS_STOP = 'busStop';
+
+    const TYPE_TRAIN_STATION = 'trainStation';
+
+    const TYPE_TOWN_HALL = 'townhall';
+
+    const TYPE_AIRPORT = 'airport';
+
+    /** @var string */
+    const ENDPOINT_URL = 'http://places-dsn.algolia.net/1/places/query';
```
Comment (Nyholm) [inline src/Provider/AlgoliaPlaces/AlgoliaPlaces.php]: Do we really want this? If you do not mind I suggest to remove it.

File: src/Provider/AlgoliaPlaces/AlgoliaPlaces.php [hunk 2]
```diff
@@ -0,0 +1,223 @@
...
+            self::TYPE_COUNTRY,
+            self::TYPE_ADDRESS,
+            self::TYPE_BUS_STOP,
+            self::TYPE_TRAIN_STATION,
+            self::TYPE_TOWN_HALL,
+            self::TYPE_AIRPORT,
+        ];
+    }
+
+    protected function getUrlContents(string $url): string
```
Comment (Nyholm) [inline src/Provider/AlgoliaPlaces/AlgoliaPlaces.php]: Coudn't you use the method on AbstractHttpProvider?

File: src/Provider/AlgoliaPlaces/AlgoliaPlaces.php (hunk 1 above)
Comment (Fabex) [reply src/Provider/AlgoliaPlaces/AlgoliaPlaces.php]: in fact in a lot of provider there is a constant like this one

GMaps
https://github.com/geocoder-php/Geocoder/blob/master/src/Provider/GoogleMaps/GoogleMaps.php#L37

For OSM it's a little different but near the same way
https://github.com/geocoder-php/Geocoder/blob/master/src/Provider/GoogleMaps/GoogleMaps.php#L37

TomTom
https://github.com/geocoder-php/Geocoder/blob/master/src/Provider/GoogleMaps/GoogleMaps.php#L37

So I think we can keep it

Comment (Nyholm) [reply src/Provider/AlgoliaPlaces/AlgoliaPlaces.php]: I mean we should not use HTTP but only HTTPS. Im fine with you using a constant.

File: src/Provider/AlgoliaPlaces/AlgoliaPlaces.php (hunk 2 above)
Comment (Fabex) [reply src/Provider/AlgoliaPlaces/AlgoliaPlaces.php]: In fact in the AbstractHttpProvider the request make a GET request, but for Algolia we need to make a POST request.

Comment (Nyholm) [reply src/Provider/AlgoliaPlaces/AlgoliaPlaces.php]: Check the latest changes (made yesterday) in `AbstractHttpProvider`. There is a new `getRequest` function that you want to override.

File: src/Provider/AlgoliaPlaces/AlgoliaPlaces.php (hunk 1 above)
Comment (Fabex) [reply src/Provider/AlgoliaPlaces/AlgoliaPlaces.php]: oh, ok. I was not understood your question. So yes I'm totally agree with I will remove it

File: src/Provider/AlgoliaPlaces/AlgoliaPlaces.php (hunk 2 above)
Comment (Fabex) [reply src/Provider/AlgoliaPlaces/AlgoliaPlaces.php]: Indeed, you have split the getUrlContents in 2 parts. So of course now I will just override getRequest and stop to duplicate some code 😄.

Comment (Nyholm) [reply src/Provider/AlgoliaPlaces/AlgoliaPlaces.php]: Yes. Correct.

**Comment by Baachi** [comment]
@Nyholm Looks good to me, can i merge them?

**Comment by Nyholm** [comment]
It does look good. But it still missing some tests, right?

**Comment by jbelien** [comment]
There is indeed no test yet in this PR.
I think it's because of https://github.com/Fabex/geocoder-algolia-places-provider/pull/1

@Fabex @sheub Can you include your tests in this PR ? Thanks :)

**Comment by Fabex** [comment]
Hi @Nyholm 
Tests are finally done. Big thanks to @sheub for his work.
I hope it's good for you

**Comment by jbelien** [comment]
Awesome !
Thanks a lot guys !

I'll have a look ASAP 👍

**Pull request merged**